File: algorithms-scala/src/EuclideanAlgorithm.scala
object EuclideanAlgorithm {
  def main(args: Array[String]): Unit = {
    print(gcd(252, 105))
  }

  @scala.annotation.tailrec
  def gcd(p: Int, q: Int): Int =
    if (q == 0)
      p
    else
      gcd(q, p % q)
}

File: algorithms-java/src/EuclideanAlgorithm.java
public class EuclideanAlgorithm {
    public static void main(String[] args) {
        System.out.println(gcd(252, 105));
    }

    public static int gcd(int p, int q) {
        if (q == 0) {
            return p;
        } else {
            return gcd(q, p % q);
        }
    }
}
